AutoCollection/diagnostic-channel/initialization.ts

```diff
@@ -7,13 +7,14 @@ import * as DiagChannelPublishers from "diagnostic-channel-publishers";
 import * as DiagChannel from "diagnostic-channel";
 import { AsyncScopeManager } from "../AsyncHooksScopeManager";
 import Logging = require("../../Library/Logging");
+import { CustomConfig } from "../../Library/CustomConfig";
 
-export const IsInitialized = !process.env["APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL"];
+export const IsInitialized = !CustomConfig.generateConfigurationObject().noDiagnosticChannel;

```

Comment:
Generation of configuration object must be a one time thing, you are actually generating every time you use it

Reply:
nice catch -- I didn't see that...

Reply:
I could be wrong -- but I meant to return `_config` object if it already exists, so it's only generated once.